crates/cairo-profiler/src/profile_builder.rs

```diff
@@ -66,32 +66,74 @@ impl ProfilerContext {
         }
     }
 
-    fn location_id(&mut self, location: &Function) -> LocationId {
-        let location = match location {
-            Function::Entrypoint(function_name) | Function::NonInlined(function_name) => {
-                function_name
+    fn locations_ids(&mut self, call_stack: &[Function]) -> Vec<LocationId> {
+        let mut locations_ids = vec![];
+        // This vector represent stacks of functions corresponding to single locations.
```

Comment:
This is bcs in pprof inlining is signalised by locations having multiple lines. So e.g. stack like that
`[A, B_inlined, C, D, E_inlined, F_inlined]` will be converted to 3 pprof locations corresponding to the following stacks:
- `[A, B_inlined]`
- `[C]`
- `[D, E_inlined, F_inlined]`